sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/RuleExecutor.scala

```diff
@@ -156,7 +163,7 @@ abstract class RuleExecutor[TreeType <: TreeNode[_]] extends Logging {
           // Only log if this is a rule that is supposed to run more than once.
           if (iteration != 2) {
             val message = s"Max iterations (${iteration - 1}) reached for batch ${batch.name}"
-            if (Utils.isTesting) {
+            if (Utils.isTesting || batch.strategy.errorOnExceed) {
               throw new TreeNodeException(curPlan, message, null)
```

Comment:
In the message, we need to explicitly tell users how to resolve this exception. Increasing the value of SQLConf.ANALYZER_MAX_ITERATIONS.key

Reply:
IMO, here is the common logic in `RuleExecutor` to handle `errorOnExceed` strategy, which should not include specific knowledge `ANALYZER_MAX_ITERATIONS` of inheritance class `analyzer`.  This will lead to potential bug when we introduce more MAX_ITERATION setting?

Maybe we can use a common phrase here like 'increasing the value of corresponding SQLConf setting'. Or an alternative way of adding a string hint field in `Strategy`, and set it in `Analyzer` together with `errorOnExceed` flag?

What do you think ? @gatorsmile  Thanks a lot!

Reply:
We need a clear message for end users. Usability is more important. 

Regarding the implementation, I think we should avoid over-engineering. Just make it simple. Adding the message postfix in abstract class Strategy looks fine to me.

Reply:
I see, thanks for the clarification. :-)